scalding-core/src/main/java/com/twitter/scalding/tap/SizeableHfs.java

```diff
@@ -0,0 +1,48 @@
+package com.twitter.scalding.tap;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.OutputCollector;
+import org.apache.hadoop.mapred.RecordReader;
+
+import cascading.scheme.Scheme;
+import cascading.tap.hadoop.Hfs;
+
+public class SizeableHfs extends Hfs {
+  public SizeableHfs(Scheme<JobConf, RecordReader, OutputCollector, ?, ?> scheme) {
+    super(scheme);
+  }
+
+  public SizeableHfs(Scheme<JobConf, RecordReader, OutputCollector, ?, ?> scheme, String stringPath) {
+    super(scheme, stringPath);
+  }
+
+  @Override
+  public long getSize(JobConf conf) throws IOException {
+    return getSize(getPath(), conf);
+  }
+
+  /**
+   * Get the total size of the file(s) specified by the Hfs, which may contain a glob
+   * pattern in its path, so we must be ready to handle that case.
+   */
+  public static long getSize(Path path, JobConf conf) throws IOException {
+    FileSystem fs = path.getFileSystem(conf);
+    FileStatus[] statuses = fs.globStatus(path);
+
+    if (statuses != null) {
+      long size = 0;
+      for (FileStatus status : statuses) {
+        size += fs.getContentSummary(status.getPath()).getLength();
+      }
+      return size;
+    } else {
```

Comment:
nit: May be do this to avoid indentation to the right:

if (status == null) {
  throw new FileNotFoundException...
}

long size = 0;
...